File: sdks/php/test/Api/AccountApiTest.php
<?php

declare(strict_types=1);

namespace Dropbox\Sign\Test\Api;

use Dropbox\Sign\Api;
use Dropbox\Sign\ApiException;
use Dropbox\Sign\Configuration;
use Dropbox\Sign\Model;
use Dropbox\Sign\Test\SignTestCase;
use Dropbox\Sign\Test\TestUtils;
use GuzzleHttp;

class AccountApiTest extends SignTestCase
{
    /** @var Api\AccountApi */
    protected $api;

    protected function setUp(): void
    {
        parent::setUp();

        $this->client = new GuzzleHttp\Client([
            'handler' => GuzzleHttp\HandlerStack::create($this->handler),
        ]);

        $this->api = new Api\AccountApi(
            Configuration::getDefaultConfiguration(),
            $this->client,
        );
    }

    public function testHttpCodeRange()
    {
        $requestClass = Model\AccountVerifyRequest::class;
        $requestData = TestUtils::getFixtureData($requestClass)['default'];

        $responseClass = Model\ErrorResponse::class;
        $responseData = TestUtils::getFixtureData($responseClass)['default'];

        $this->setExpectedResponse($responseData, mt_rand(400, 499));

        $obj = Model\AccountVerifyRequest::init($requestData);

        $error = null;

        try {
            $this->api->accountVerify($obj);
        } catch (ApiException $e) {
            $error = $e->getResponseObject();
        }

        $this->assertInstanceOf($responseClass, $error);
        $this->assertEquals($responseData, TestUtils::toArray($error));
    }

    public function testAccountCreate()
    {
        $requestClass = Model\AccountCreateRequest::class;
        $requestData = TestUtils::getFixtureData($requestClass)['default'];

        $responseClass = Model\AccountCreateResponse::class;
        $responseData = TestUtils::getFixtureData($responseClass)['default'];

        $this->setExpectedResponse($responseData);

        $obj = Model\AccountCreateRequest::init($requestData);

        $response = $this->api->accountCreate($obj);

        $this->assertInstanceOf($responseClass, $response);
        $this->assertEquals($responseData, TestUtils::toArray($response));
    }

    public function testAccountGet()
    {
        $responseClass = Model\AccountGetResponse::class;
        $responseData = TestUtils::getFixtureData($responseClass)['default'];

        $this->setExpectedResponse($responseData);

        $response = $this->api->accountGet();

        $this->assertInstanceOf($responseClass, $response);
        $this->assertEquals($responseData, TestUtils::toArray($response));
    }

    public function testAccountUpdate()
    {
        $requestClass = Model\AccountUpdateRequest::class;
        $requestData = TestUtils::getFixtureData($requestClass)['default'];

        $responseClass = Model\AccountGetResponse::class;
        $responseData = TestUtils::getFixtureData($responseClass)['default'];

        $this->setExpectedResponse($responseData);

        $obj = Model\AccountUpdateRequest::init($requestData);

        $response = $this->api->accountUpdate($obj);

        $this->assertInstanceOf($responseClass, $response);
        $this->assertEquals($responseData, TestUtils::toArray($response));
    }

    public function testAccountVerify()
    {
        $requestClass = Model\AccountVerifyRequest::class;
        $requestData = TestUtils::getFixtureData($requestClass)['default'];

        $responseClass = Model\AccountVerifyResponse::class;
        $responseData = TestUtils::getFixtureData($responseClass)['default'];

        $this->setExpectedResponse($responseData);

        $obj = Model\AccountVerifyRequest::init($requestData);

        $response = $this->api->accountVerify($obj);

        $this->assertInstanceOf($responseClass, $response);
        $this->assertEquals($responseData, TestUtils::toArray($response));
    }
}

File: sdks/java-v1/src/test/java/com/dropbox/sign/api/AccountApiTest.java
package com.dropbox.sign.api;

import com.dropbox.sign.ApiClient;
import com.dropbox.sign.ApiException;
import com.dropbox.sign.TestHelper;
import com.dropbox.sign.model.*;
import com.fasterxml.jackson.databind.JsonNode;
import org.junit.Assert;
import org.junit.Test;

import java.util.stream.IntStream;

public class AccountApiTest {
    @Test
    public void accountCreateTest() throws Exception {
        JsonNode expectedResponseData = TestHelper.getJsonContents(AccountCreateResponse.class.getSimpleName());
        AccountCreateResponse expectedResponse = AccountCreateResponse.init(expectedResponseData.toString());
        ApiClient apiClient = TestHelper.setUpMock(200, expectedResponse);

        JsonNode requestData = TestHelper.getJsonContents(AccountCreateRequest.class.getSimpleName());

        AccountCreateRequest request = AccountCreateRequest.init(requestData.toString());

        AccountApi accountApi = new AccountApi(apiClient);
        AccountCreateResponse response = accountApi.accountCreate(request);

        Assert.assertEquals(expectedResponse, response);
    }

    @Test
    public void accountGetTest() throws Exception {
        JsonNode expectedResponseData = TestHelper.getJsonContents(AccountGetResponse.class.getSimpleName());
        AccountGetResponse expectedResponse = AccountGetResponse.init(expectedResponseData.toString());
        ApiClient apiClient = TestHelper.setUpMock(200, expectedResponse);

        AccountApi accountApi = new AccountApi(apiClient);
        AccountGetResponse response = accountApi.accountGet(null, "jack@example.com");

        Assert.assertEquals(expectedResponse, response);
    }

    @Test
    public void accountUpdateTest() throws Exception {
        JsonNode expectedResponseData = TestHelper.getJsonContents(AccountGetResponse.class.getSimpleName());
        AccountGetResponse expectedResponse = AccountGetResponse.init(expectedResponseData.toString());
        ApiClient apiClient = TestHelper.setUpMock(200, expectedResponse);

        JsonNode requestData = TestHelper.getJsonContents(AccountUpdateRequest.class.getSimpleName());

        AccountUpdateRequest request = AccountUpdateRequest.init(requestData.toString());

        AccountApi accountApi = new AccountApi(apiClient);
        AccountGetResponse response = accountApi.accountUpdate(request);

        Assert.assertEquals(expectedResponse, response);
    }

    @Test
    public void accountVerifyTest() throws Exception {
        JsonNode expectedResponseData = TestHelper.getJsonContents(AccountVerifyResponse.class.getSimpleName());
        AccountVerifyResponse expectedResponse = AccountVerifyResponse.init(expectedResponseData.toString());
        ApiClient apiClient = TestHelper.setUpMock(200, expectedResponse);

        JsonNode requestData = TestHelper.getJsonContents(AccountVerifyRequest.class.getSimpleName());

        AccountVerifyRequest request = AccountVerifyRequest.init(requestData.toString());

        AccountApi accountApi = new AccountApi(apiClient);
        AccountVerifyResponse response = accountApi.accountVerify(request);

        Assert.assertEquals(expectedResponse, response);
    }

    @Test
    public void testHttpCodeRange() throws Exception {
        JsonNode expectedResponseData = TestHelper.getJsonContents(ErrorResponse.class.getSimpleName());
        ErrorResponse expectedResponse = ErrorResponse.init(expectedResponseData.toString());

        JsonNode requestData = TestHelper.getJsonContents(AccountVerifyRequest.class.getSimpleName());

        AccountVerifyRequest request = AccountVerifyRequest.init(requestData.toString());

        IntStream.range(400, 500).forEach(value -> {
            try {
                ApiClient apiClient = TestHelper.setUpMock(value, expectedResponse);
                AccountApi accountApi = new AccountApi(apiClient);
                accountApi.accountVerify(request);
                Assert.fail();
            } catch (ApiException e) {
                Assert.assertEquals(value, e.getCode());
                Assert.assertEquals(expectedResponse, e.getErrorResponse());
            }
        });
    }
}
